app/gui/src/dashboard/components/AriaComponents/IconDisplay/IconDisplay.tsx

```diff
@@ -0,0 +1,81 @@
+/** @file A text display with an icon. */
+import { Icon } from '#/components/Icon'
+import { tv, type VariantProps } from '#/utilities/tailwindVariants'
+import { Text, WithVisualTooltip, type IconProp, type TextProps, type TooltipElementType } from '..'
+
+const ICON_DISPLAY_STYLES = tv({
+  base: 'flex items-center gap-2 max-w-48 min-w-4 px-1',
+  slots: {
+    icon: '-mb-0.5',
+    // For some reason `min-w-0` is required for the ellipsis to appear.
+    container: 'flex mx-auto min-w-0',
+    text: 'block truncate',
+  },
+  variants: {
+    variant: {
+      custom: '',
+      link: 'inline-block px-0 py-0 rounded-sm text-primary/50 underline border-0',
+      primary: 'bg-primary text-white',
+      accent: 'bg-accent text-white',
+      ghost: 'text-primary',
+      submit: 'bg-invite text-white opacity-80',
+      outline: 'border-0.5 rounded-full border-primary/20 text-primary',
+    },
+  },
+  defaultVariants: {
+    variant: 'custom',
+  },
+})
+
+/** Render props for {@link IconDisplay}. */
+export interface IconDisplayRenderProps {
+  /** Defaults to `true`. */
+  readonly isCurrent?: boolean
+  /** Defaults to `false`. */
+  readonly isDisabled?: boolean
+}
+
+/** Props for an {@link IconDisplay}. */
+export interface IconDisplayProps<IconType extends string>
+  extends Omit<TextProps, 'children' | 'variant' | 'variants'>,
+    IconDisplayRenderProps,
+    VariantProps<typeof ICON_DISPLAY_STYLES> {
+  readonly icon: IconProp<IconType, Required<IconDisplayRenderProps>>
+  readonly children:
+    | TooltipElementType
+    | ((renderProps: Required<IconDisplayRenderProps>) => TooltipElementType)
+}
+
+/** A text display with an icon. */
+export function IconDisplay<IconType extends string>(props: IconDisplayProps<IconType>) {
```

Comment:
when I told about `patterns` I meant only the `ICON_DISPLAY_STYLES` part,

Reply:
i guess, but i do want a dedicated component to avoid having to rewrite the markup every time, unless you have a better idea